src/planscape/forsys/raster_condition_fetcher.py

```diff
@@ -51,87 +73,107 @@ class RasterConditionFetcher:
     #       value exists for a given pixel, the element corresponding to the
     #       pixel is a float; otherwise, it's np.nan.
     #       note: the value, for now, is 1.0 - normalized condition value.
+    #   - <attribute name>: each attribute (e.g. slope, buildings) has its own
+    #       column. If a priority value exists for a given pixel, the element
+    #       corresponding to the pixel is a float; otherwise, it's np.nan.
     data: dict[str, list]
     # Maps x-pixel to y-pixel to a row index in the self.data dataframe.
     x_to_y_to_index: dict[int, dict[int, int]]
 
-    def __init__(self, region: str, priorities: list[str], geo: GEOSGeometry):
+    def __init__(
+            self, region: str, priorities: list[str],
+            land_attributes: list[str],
+            geo: GEOSGeometry):
         raster_geo = get_raster_geo(geo)
 
         conditions = get_conditions(region, priorities)
-        self.conditions_to_raster_values = \
-            self._fetch_conditions_to_raster_values(
-                conditions, raster_geo)
+        attributes = get_attributes(land_attributes)
+        self.raster_values = \
+            self._fetch_raster_values(conditions, attributes, raster_geo)
 
         self.topleft_coords = self._get_topleft_coords(
-            self.conditions_to_raster_values)
+            self.raster_values)
         self.data, self.x_to_y_to_index = self._reformat_to_dataframe(
-            self.conditions_to_raster_values, priorities)
+            self.raster_values, self.topleft_coords, priorities, land_attributes)
         self.width, self.height = self._get_width_and_height(
             self.x_to_y_to_index)
-        
-        # TODO: Only stands with at least one condition value are saved; 
-        # however, some stands may have 0 condition values, but still be useful 
-        # to include in the forsys input (think stand threshold, EPW). These 
+
+        # TODO: Only stands with at least one condition value are saved;
+        # however, some stands may have 0 condition values, but still be useful
+        # to include in the forsys input (think stand threshold, EPW). These
         # should be added to self.data.
 
+    def _fetch_raster_values(self,
+                             conditions, attributes, raster_geo) -> dict[str, RasterPixelValues]:
+        raster_values = {}
+        self._fetch_condition_raster_values(
+            conditions, raster_geo, raster_values)
+        self._fetch_attribute_raster_values(
+            attributes, raster_geo, raster_values)
+        return raster_values
+
     # Fetches condition raster values for a given GEOSGeometry and emits it in
     # a {condition name: ConditionPixelValues} dictionary.
```

Comment:
update to RasterPixelValues.

Reply:
Done.
I've fixed this throughout the file (oops).